static/app/utils/replays/hooks/useScrollPosition.tsx

```diff
@@ -0,0 +1,82 @@
+import {useEffect, useRef} from 'react';
+
+import {
+  ScrollContextPositionTab,
+  useScrollContext,
+} from 'sentry/components/replays/scrollContext';
+import usePrevious from 'sentry/utils/usePrevious';
+
+type Options = {
+  id: keyof typeof ScrollContextPositionTab;
+  scrollContainerRef: React.RefObject<HTMLElement>;
+};
+
+export function useScrollPosition({scrollContainerRef, id}: Options) {
+  const {scrollPosition, setScrollPosition} = useScrollContext();
+  const previousScrollPositions = usePrevious(scrollPosition);
+  const mountedRef = useRef(false);
+
+  // Save the last scroll position in the state
+  useEffect(() => {
+    const scrollContainer = scrollContainerRef.current;
+
+    const handleScroll = () => {
+      if (scrollContainer) {
+        setScrollPosition(prevScrollPosition => {
+          const currentScrollPosition = scrollContainer?.scrollTop ?? 0;
+          const existingScrollPosition = prevScrollPosition.find(
+            position => position.id === id
+          );
+
+          if (existingScrollPosition) {
+            existingScrollPosition.scrollPosition = currentScrollPosition;
+          } else {
+            prevScrollPosition.push({
+              id,
+              scrollPosition: currentScrollPosition,
+            });
+          }
```

Comment:
You could probably simplify this logic by just initializing each tab to 0. Don't think you'll need to check the prev value on scroll can just set it